packages/core/lib/attributes.ts

```diff
@@ -60,9 +60,12 @@ export function attributeToJson (key: string, attribute: SpanAttribute): JsonAtt
       if (Number.isNaN(attribute) || !Number.isFinite(attribute)) {
         return undefined
       }
-      if (Number.isInteger(attribute)) {
+
+      // 'bugsnag.sampling.p' must always be sent as a doubleValue
+      if (key !== 'bugsnag.sampling.p' && Number.isInteger(attribute)) {
```

Comment:
This is a bit of a hack but we don't have a way to tell the attribute encoding that something needs to be of a specific type

Maybe this kind of auto-encoding was a bad idea?

Reply:
It definitely made sense when we initially implemented it!

Not too concerned about it - but worth refactoring if we end up adding _more_ edge cases, I'd say?